scaleway/resource_domain_record.go

```diff
@@ -0,0 +1,426 @@
+		return nil
+	}
+
+	d.SetId(record.ID)
+	_ = d.Set("data", flattenDomainData(record.Data, record.Type))
+	_ = d.Set("ttl", int(record.TTL))
+	_ = d.Set("priority", int(record.Priority))
+	_ = d.Set("geo_ip", flattenDomainGeoIP(record.GeoIPConfig))
+	_ = d.Set("http_service", flattenDomainHTTPService(record.HTTPServiceConfig))
+	_ = d.Set("weighted", flattenDomainWeighted(record.WeightedConfig))
+	_ = d.Set("view", flattenDomainView(record.ViewConfig))
+
+	return nil
+}
+
+func resourceScalewayDomainRecordUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	domainAPI := newDomainAPI(meta)
+
+	if d.HasChangesExcept("dns_zone", "keep_empty_zone", "name", "type") {
+		id := d.Id()
+
+		geoIP, okGeoIP := d.GetOk("geo_ip")
+		_, err := domainAPI.UpdateDNSZoneRecords(&domain.UpdateDNSZoneRecordsRequest{
+			DNSZone: d.Get("dns_zone").(string),
+			Changes: []*domain.RecordChange{
+				{
+					Set: &domain.RecordChangeSet{
+						ID: &id,
+						Records: []*domain.Record{
+							{
+								Name:              d.Get("name").(string),
+								Data:              d.Get("data").(string),
+								Priority:          uint32(d.Get("priority").(int)),
+								TTL:               uint32(d.Get("ttl").(int)),
+								Type:              domain.RecordType(d.Get("type").(string)),
+								GeoIPConfig:       expandDomainGeoIPConfig(d.Get("data").(string), geoIP, okGeoIP),
+								HTTPServiceConfig: expandDomainHTTPService(d.GetOk("http_service")),
+								WeightedConfig:    expandDomainWeighted(d.GetOk("weighted")),
+								ViewConfig:        expandDomainView(d.GetOk("view")),
+							},
+						},
+					},
+				},
+			},
+			ReturnAllRecords: scw.BoolPtr(false),
+		})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return resourceScalewayDomainRecordRead(ctx, d, meta)
+}
+
+func resourceScalewayDomainRecordDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	domainAPI := newDomainAPI(meta)
+
+	id := d.Id()
+	_, err := domainAPI.UpdateDNSZoneRecords(&domain.UpdateDNSZoneRecordsRequest{
+		DNSZone: d.Get("dns_zone").(string),
+		Changes: []*domain.RecordChange{
+			{
+				Delete: &domain.RecordChangeDelete{
+					ID: &id,
+				},
+			},
+		},
+		ReturnAllRecords: scw.BoolPtr(false),
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId("")
+
+	// if the zone have only NS records, then delete the all zone
+	if !d.Get("keep_empty_zone").(bool) {
+		res, err := domainAPI.ListDNSZoneRecords(&domain.ListDNSZoneRecordsRequest{
+			DNSZone: d.Get("dns_zone").(string),
+		}, scw.WithAllPages())
+
+		if err != nil {
+			if is404Error(err) {
+				return nil
+			}
+			return diag.FromErr(err)
+		}
+
+		hasRecords := false
+		for _, r := range res.Records {
+			if r.Type != domain.RecordTypeNS {
+				hasRecords = true
+				break
+			}
+		}
+
+		if !hasRecords {
+			_, err = domainAPI.DeleteDNSZone(&domain.DeleteDNSZoneRequest{
+				DNSZone: d.Get("dns_zone").(string),
```

Comment:
Please add project_id in the request. It is mandatory in the `DeleteDNSZoneRquest`